packages/asset/contracts/LazyVault.sol

```diff
@@ -0,0 +1,131 @@
+import {ILazyVault} from "./interfaces/ILazyVault.sol";
+
+/// @title LazyVault
+/// @author The Sandbox
+/// @notice Contract responsible for distributing tokens to creators and split recipients
+contract LazyVault is ILazyVault, AccessControl {
+    bytes32 public constant MANAGER_ROLE = keccak256("MANAGER_ROLE");
+    bytes32 public constant DISTRIBUTOR_ROLE = keccak256("DISTRIBUTOR_ROLEw");
+
+    // address of the token that will be distributed by this vault
+    address public vaultToken;
+    // the value of each tier
+    uint256[] public tierValues;
+
+    // array of recipients and their split
+    SplitRecipient[] public splitRecipients;
+
+    /// @notice Sets the initial vault token, tier values, and split recipients
+    /// @param _tierValues The initial tier values
+    /// @param _initialRecipients The initial split recipients
+    /// @param _vaultToken The vault token
+    constructor(
+        address _admin,
+        address _manager,
+        address _distributor,
+        uint256[] memory _tierValues,
+        SplitRecipient[] memory _initialRecipients,
+        address _vaultToken
+    ) {
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+        _grantRole(MANAGER_ROLE, _manager);
+        _grantRole(DISTRIBUTOR_ROLE, _distributor);
+        _setSplitRecipients(_initialRecipients);
+        _setTierValues(_tierValues);
+        vaultToken = _vaultToken;
+    }
+
+    /// @notice Distributes tokens to the creators and shares a specified percentage with the split recipients
+    /// @dev The order of items in the tiers, amounts, and creators arrays must be preserved
+    /// @param tiers An array of tiers
+    /// @param amounts An array of amounts
+    /// @param creators An array of creators
+    function distribute(
+        uint8[] calldata tiers,
+        uint256[] calldata amounts,
+        address[] calldata creators
+    ) external onlyRole(DISTRIBUTOR_ROLE) {
+        require(tiers.length == amounts.length, "LazyVault: 1-Array mismatch");
+        require(tiers.length == creators.length, "LazyVault: 2-Array mismatch");
+
+        for (uint256 i = 0; i < tiers.length; i++) {
+            require(tiers[i] > 0 && tiers[i] < tierValues.length, "LazyVault: Invalid tier");
+            uint256 totalValue = amounts[i] * tierValues[tiers[i]];
+            uint256 creatorShare = totalValue;
+            uint256[] memory splitValues = new uint256[](splitRecipients.length);
+            address[] memory splitRecipientsAddresses = new address[](splitRecipients.length);
+            for (uint256 j = 0; j < splitRecipients.length; j++) {
+                SplitRecipient memory recipient = splitRecipients[j];
+                splitRecipientsAddresses[j] = recipient.recipient;
+                uint256 splitValue = (totalValue * recipient.split) / 10000;
+                splitValues[j] = splitValue;
+                IERC20(vaultToken).transfer(recipient.recipient, splitValue);
```

Comment:
should use `SafeERC20` for transfers https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/utils/SafeERC20.sol